python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -156,17 +220,68 @@ def __init__(
                 'Upgrade with: `pip install -U "xgboost>=1.7"`'
             )
 
-        # TODO(justinvyu): [Deprecated] Remove in 2.11
-        if dmatrix_params != _DEPRECATED_VALUE:
-            raise DeprecationWarning(
-                "`dmatrix_params` is deprecated, since XGBoostTrainer no longer "
-                "depends on the `xgboost_ray.RayDMatrix` utility. "
-                "You can remove this argument and use `dataset_config` instead "
-                "to customize Ray Dataset ingestion."
+        # TODO(justinvyu): [Deprecated] Legacy XGBoostTrainer API
+        legacy_api = train_loop_per_worker is None
+        if legacy_api:
+            train_loop_per_worker = self._get_legacy_train_fn_per_worker(
+                xgboost_train_kwargs=train_kwargs,
+                run_config=run_config,
+                label_column=label_column,
+                num_boost_round=num_boost_round,
+                datasets=datasets,
             )
+            train_loop_config = params or {}
+        elif train_kwargs:
```

Comment:
Should this also check the other args?
```suggestion
        elif train_kwargs or label_column or params or num_boost_rounds:
```